### ISSUE TITLE
Error: Cannot find module 'date-fns/isValid'

### ISSUE THREAD

**Issue by juan55860** [mikeerickson/validatorjs#350]
Hello guys

Using node 8.x and 10.x I have this error 

```
Error: Cannot find module 'date-fns/isValid'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
    at Function.Module._load (internal/modules/cjs/loader.js:508:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/MyProject/node_modules/validatorjs/src/rules.js:1:77)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
```

Please check [https://www.npmjs.com/package/date-fns](url)

Actually, I solved the problem using  @date-fns/upgrade in my project.

`npm install @date-fns/upgrade --save
`

Thanks a lot 👍

**Comment by john-aws** [comment]
The bigger question is why did validatorjs drop its explicit dependency on `date-fns`, in the process potentially breaking any project with dependency `"validatorjs": "^3.17.1"` ?

Until today, my package-lock.json contained:

```
"validatorjs": {
  "version": "3.17.1",
  "resolved": "https://registry.npmjs.org/validatorjs/-/validatorjs-3.17.1.tgz",
  "integrity": "<cut>",
  "requires": {
    "date-fns": "2.0.1"
  }
},
```

As of today (validatorjs@3.18.0), it reads:

```
"validatorjs": {
  "version": "3.18.0",
  "resolved": "https://registry.npmjs.org/validatorjs/-/validatorjs-3.18.0.tgz",
  "integrity": "<cut>"
},
```

I see it's related to an attempt to reduce the size of the `date-fns` import per https://github.com/skaterdav85/validatorjs/commit/369d5663761624c77739ee85613c87f7efc98de4

**Comment by sanonz** [comment]
At file: https://unpkg.com/browse/validatorjs@3.18.0/package.json

```json
{
    "dependencies": {},
    "devDependencies": {
        "browserify": "^13.0.1",
        "chai": "3.5.0",
        "date-fns": "2.1.0",
        // ...
    }
}
```

Correctly
```json
{
    "dependencies": {
        "date-fns": "2.1.0"
    },
    "devDependencies": {
        "browserify": "^13.0.1",
        "chai": "3.5.0",
        // ...
    }
}
```

**Comment by john-aws** [comment]
So, it looks like it went from being a [runtime dependency in 3.17.1](https://unpkg.com/browse/validatorjs@3.17.1/package.json) to a [dev dependency in 3.18.0](https://unpkg.com/browse/validatorjs@3.18.0/package.json).

**Comment by mikeerickson** [comment]
Sorry for delay on this, I will get this cured ASAP and post a fix. Not sure where the regression occurred, but I will remedy it. In the meantime, installing `date-fns` manually will cure

**Comment by juan55860** [comment]
@mikeerickson great, thanks a lot for your help, and all by your interest.

Cheers.

**Comment by mikeerickson** [comment]
I have patched the 3.18 release (verison 3.18.1) which fixes this regression.
